swing/src/net/sf/openrocket/gui/main/SimulationPanel.java

```diff
@@ -598,6 +624,216 @@ public void updateEnabledState() {
 		}
 	}
 
+	class DumpSimulationToCSVAction extends SimulationAction {
+		private String lastSelectedLocation = "";
+		private HashMap<String, String> cToUnit;
+		public DumpSimulationToCSVAction() {
+			putValue(NAME, trans.get("simpanel.pop.export_to_csv"));
+			putValue(SMALL_ICON, Icons.FILE_EXPORT_AS);
+		}
+
+		/**
+		 * Means by wich the CSV export will clean up units on the values and
+		 * describe them on the header fields instead.
+		 */
+		private void populateColumnNameToUnitsHashTable() {
+			if (null == simulationTableModel) {
+				return;
+			}
+			valueColumnToUnitString.clear(); // necessary if units changed during session
+			for (int i=0; i<simulationTableModel.getColumnCount(); i++) {
+				Column c = simulationTableModel.getColumn(i);
+				if (c instanceof ValueColumn) {
+					// only value columns seem to have units that are not zero length strings... these are
+					// the ones we actually want in our lookup table.
+					valueColumnToUnitString.put(c.toString(), c.getUnits().getDefaultUnit().getUnit());
+				}
+				
+			}
+		}
+		/**
+		 * Dump data from sim table to file for run simulations
+		 * @param data The csv data as one string block.
+		 * @param csvFile The file to dump the data to.
+		 */
+		private void dumpDataToFile(String data, File csvFile) {
+			BufferedWriter bufferedWriter = null;
+			try {
+				csvFile.createNewFile();
+				bufferedWriter = new BufferedWriter(new FileWriter(csvFile));
+				bufferedWriter.write(data);
+				this.lastSelectedLocation = csvFile.getParent();
+			} catch (FileNotFoundException e) {
+				String msg = e.getMessage();
+				JOptionPane.showMessageDialog(simulationTable.getParent(), msg);
+			} catch (IOException e) {
+				String msg = e.getMessage();
+				JOptionPane.showMessageDialog(simulationTable.getParent(), msg);
+			} finally {
+				if (bufferedWriter != null) {
+					try {
+						bufferedWriter.close();
+					} catch (IOException e) {
+						String msg = e.getMessage();
+						JOptionPane.showMessageDialog(simulationTable.getParent(), msg);
+					}
+				}
+			}
+		}
+
+		private JFileChooser setUpFileChooser() {
+			JFileChooser fch = new JFileChooser();
+			String saveDialogTitle = trans.get("simpanel.pop.export_to_csv.save.dialog.title");
+			String saveButtonText = trans.get("simpanel.pop.export_to_csv.save.button.text");
+			fch.setApproveButtonText(saveButtonText);
+			fch.setDialogTitle(saveDialogTitle);
+			fch.setApproveButtonToolTipText(saveDialogTitle);
+			// set up to filter for .csv's
+			fch.setFileFilter(new FileFilter() {
+				@Override
+				public boolean accept(File f) {
+					if (f.isDirectory()) {
+						return true;
+					} else {
+						return f.getName().toLowerCase().endsWith(".csv");
+					}
+				}
+
+				@Override
+				public String getDescription() {
+					return trans.get("FileHelper.CSV_FILTER");
+				}
+				
+			});
+
+			// default output csv to same name as "the rocket".
```

Comment:
`// Default output CSV to the rocket name`
Your comment otherwise makes it seem like you would actually name it "the rocket", instead of the rocket's name (try to reserve double-quotes in your comments only for Strings)

Reply:
a matter of temperament on my part.